test/auth_cluster/test_maintenance_socket.py

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright (C) 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, NoHostAvailable
+from cassandra import Unauthorized
+from cassandra.connection import UnixSocketEndPoint
+from test.pylib.manager_client import ManagerClient
+
+import pytest
+
+@pytest.mark.xfail(reason="regression in python driver (https://github.com/scylladb/python-driver/issues/278)")
+@pytest.mark.asyncio
+async def test_maintenance_socket(manager: ManagerClient):
+    """
+    Test that when connecting to the maintenance socket, the user has superuser permissions,
+    even if the authentication is enabled on the regular port.
+    """
+    config = {
+        "authenticator": "PasswordAuthenticator",
+        "authorizer": "CassandraAuthorizer",
+    }
+
+    server = await manager.server_add(config=config)
+    workdir = await manager.server_get_workdir(server.server_id)
+    socket = workdir + "/cql.maintenance"
```

Comment:
Looks like the test is outdated now.
But I think we can leave fixing it until after the Python driver issue is fixed and new version of the driver is released.